### ISSUE TITLE
 fatal error: termios.h: No such file or directory

### ISSUE THREAD

**Issue by trsdln** [odeke-em/drive#360]
while running 'go get -u github.com/odeke-em...' got:
```
~ $ go get -u github.com/odeke-em/drive/cmd/drive
# github.com/odeke-em/drive/src
.gocode/src/github.com/odeke-em/drive/src/tty.go:20:21: fatal error: termios.h: No such file or directory
 #include <termios.h>
                     ^
compilation terminated.
```

**Comment by odeke-em** [comment]
Hello there @trsdln, Thank you for reporting this and welcome to drive!

For starters, could you provide your OS details, and Go version?

**Comment by odeke-em** [comment]
If using Windows please see https://github.com/odeke-em/drive/wiki/Building-on-Windows.
Although drive most likely won't work on Windows as it is a *NIX built program but maybe in the future.

**Comment by trsdln** [comment]
@odeke-em, 
OS: Linux Mint 17.2 Rafaela Cinnamon
Go version: 1.5 linux/amd64

**Comment by odeke-em** [comment]
Thanks for that information @trsdln. That's odd though. Would you mind reinstalling Go if it is possible?
Also before installing would you mind testing out like this
```shell
$ CGO_ENABLED=0 go get -u github.com/odeke-em/drive/cmd/drive
```

**Comment by trsdln** [comment]
I didn't reinstall Go, but executed:
```
$ CGO_ENABLED=0 go get -u github.com/odeke-em/drive/cmd/drive
```
and it executed with success.
Thanks a lot!

Also, seems to there is some mistakes in project's `readme`:
```
#it should be:
> export GOPATH=$HOME/gopath
> export PATH=$GOPATH:$GOPATH/bin:$PATH

#instead of:
> export GOPATH=\$HOME/gopath
> export PATH=\$GOPATH:\$GOPATH/bin:\$PATH
```
If dollar sign escaped it prevents resolving of variable and inserts name of variable instead.

**Comment by odeke-em** [comment]
Glad to know that you got that fixed.

`
Also, seems to be there is some mistakes in project's readme:
`
Actually that isn't correct in the context that was provided in the README. The README includes a snippet to write code to your file. Sure if you copied and pasted it, yes but if you followed the code snippet as is, unescaping the variables will actually substitute the paths which isn't something you want and want your $GOPATH to resolved lazily and not hardcoded
<img width="783" alt="screen shot 2015-09-02 at 12 32 02 pm" src="https://cloud.githubusercontent.com/assets/4898263/9640470/080b318c-516f-11e5-8461-4071031c1565.png">


I implore you to try what you had suggested and you'll see that the entire paths are resolved and inserted into the bashrc file which isn't what we want.

Also it is odd that your Go build doesn't have cgo enabled. termios.h is a file that is needed quite a bit by lots of shell commands and C commands, but oh well that resolved it for you.

**Comment by trsdln** [comment]
ok, thanks

**Comment by mikegrima** [comment]
Hello:

Like @trsdln, I'm running Linux Mint 17.2 Rafaela Cinnamon x64, and have installed Go 1.5.1.  I also encountered the same error.  However, I also got other permission denied errors.

I was just about to write a detailed post with the error messages, but then I realized what the problem was: I didn't have ``build-essential`` installed.  Running ``$ apt-get install build-essential`` resolved my issue.

Should that be added to the installation docs?

**Comment by odeke-em** [comment]
Hello @mikegrima welcome to drive! This is great news to hear, in deed we should note this information down somewhere. Would you mind creating a wiki entry for this?

**Comment by nassausky** [comment]
Do you have any idea why after I had the error and followed your instructions above and the cursor returned to the command prompt and nothing happened? Nothing is installed in the current folder.
Thanks.

**Comment by nassausky** [comment]
Just a heads up. Installing on Windows 10 64bit.  It looks like the installation for Drive prefers 32 bit. Found out my problem where the command
`go get -u github.com/odeke-em/drive/cmd/drive`
returns back to the command prompt after a minute without any sign that it did anything.
The answer is that the Go installer isn't so great.  The least it should tell you is it was installed properly. Somewhere it should say clearly to also setup a GOPATH environmental variable with a path that you'd like to place the output files of the GIT download and compile results. Turns out it put a binary (a windows executable drive.exe) in my GOPATH after I ran the command above and I didn't see it in the current directory I was in.  That path was not in my system path so when I tried running Drive from the current location, it couldn't find it.
Hope this helps someone in the future.
